### ISSUE TITLE
Phantom rebase-merge directory (`rm -rf ".git/rebase-merge"`)

### ISSUE THREAD

**Issue by devinrhode2** [gitkraken/vscode-gitlens#1732]
```
➤ git rebase chore-lint-and-doc-changes-tweak-unknown-action
fatal: It seems that there is already a rebase-merge directory, and
I wonder if you are in the middle of another rebase.  If that is the
case, please try
	git rebase (--continue | --abort | --skip)
If that is not the case, please
	rm -fr "/Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge"
and run me again.  I am stopping in case you still have something
valuable there.

➤ git status
On branch chore-update-next-wip
Your branch and 'origin/chore-update-next-wip' have diverged,
and have 33 and 19 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

You are currently rebasing.
  (all conflicts fixed: run "git rebase --continue")

nothing to commit, working tree clean
➤ git rebase --abort
warning: could not read '/Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge/head-name': No such file or directory
➤ cat /Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge
cat: /Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge: Is a directory
➤ ls /Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge
git-rebase-todo
➤ cat /Users/me/repos/MY_APP/.git/worktrees/MY_APP2/rebase-merge/git-rebase-todo
➤ code .
```

I'll have to apologize, I do not have gitlens debug output enabled.

- GitLens Version: v11.7.0
- Git Version: 2.34.0 (very recently updated)

```
CPUs | Apple M1 (8 x 24)
-- | --
GPU Status | 2d_canvas: enabled gpu_compositing: enabled metal: disabled_off multiple_raster_threads: enabled_on oop_rasterization: enabled opengl: enabled_on rasterization: enabled skia_renderer: disabled_off_ok video_decode: enabled webgl: enabled webgl2: enabled
Load (avg) | 2, 2, 2
Memory (System) | 16.00GB (0.01GB free)
Process Argv | --crash-reporter-id aad11fcf-6f4b-42d6-a3a5-281835fd9d8b
Screen Reader | no
VM | 0%
```

Steps to Reproduce:
I will help fill these in next time the issue happens.
I am using the modern git lens rebase-todo gui, but I also use command line some.
I feel I almost never ran into this issue with git 2.33.0, so I have my suspicions there.

I suspect that when the gitlens todo gui is closed, there should be a check to see if `rebase-merge` dir has one file, `git-rebase-todo` which is empty

**Comment by devinrhode2** [comment]
I just realized, maybe I could have resolved it by running `git rebase --continue` as suggested in git status

**Comment by devinrhode2** [comment]
Actually, `git rebase --continue` gives the same error as `git rebase --abort`

**Comment by tvardero** [comment]
Can confirm that after using interactive git-rebase (either aborting or confirming rebase), folder '.git/rebase-merge' is not being deleted, but is being attempted to. 
```Deletion of directory '.git/rebase-merge' failed. Should I try again? (y/n)
> y
Deletion of directory '.git/rebase-merge' failed. Should I try again? (y/n)
> y
...
```

I have to delete it manually everytime with ```rm -fr ".git/rebase-merge"```.

**Comment by devinrhode2** [comment]
Idk how this may relate to the fancy git lens interactive rebase editor, which I love, but the save state of that buffer seems particularly buggy.

Sometimes there's no file, the buffer should be wiped. (Ideally force close the tab, but definitely No undo/redo)

Sometimes the buffer is stale. I'm in a new interactive rebase, and need to hit undo once... very strange... I get sketched out and restart my computer (exaggerating)

**Comment by devinrhode2** [comment]
I've stopped using the fancy interactive rebase editor, but I _think_ this can still happen.

It may have more to do with vscode if anything.

**Comment by eamodio** [comment]
Can you please verify this fix in tomorrow's pre-release edition of GitLens?

You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/199397587-6948b1bc-3019-4d85-a6d5-332fea5d8f78.png)

**Comment by devinrhode2** [comment]
Hey @eamodio, my apologies - I haven't run into this since Oct 27th, but that may be due to experience with git rebase and also not using gitlens's interactive rebase editor.

I did get an idea that will make the state of this `rebase-merge` dir easier to manage, independent of how vscode handles buffers: https://github.com/gitkraken/vscode-gitlens/issues/2356

TLDR: Avoid directly reading/writing from `.git/rebase-merge/git-rebase-todo`, don't let vscode control whether or not that dir is recreated on writes.
